src/core/search/search.cc

```diff
@@ -224,8 +233,9 @@ struct BasicSearch {
   // "term": access field's text index or unify results from all text indices if no field is set
   IndexResult Search(const AstTermNode& node, string_view active_field) {
     if (!active_field.empty()) {
-      auto* index = GetIndex<TextIndex>(active_field);
-      return index->Matching(node.term);
+      if (auto* index = GetIndex<TextIndex>(active_field); index)
+        return index->Matching(node.term);
+      return IndexResult{};
```

Comment:
My only only concern about this is if there is some misuse of this down the execution path. So default constructing this makes the variant to own an empty DocVec. Basically the path is:

result = SearchGeneric() -> Search variant
result then gets passed to KnnVariants or to KnnDistance

So far so good, because I think all of the algorithms used comply with these conditions.

After all that we return the SearchResult. The question here is that the SearchResult will be empty (maybe I am missing something) because of the missing index. Shouldn't that be treated as an error? That is, populate the error_ field if the index is missing?

p.s. I am probably missing something so I thought asking.